d2/src/main/java/com/linkedin/d2/balancer/simple/SimpleLoadBalancer.java

```diff
@@ -531,6 +536,11 @@ private ServiceProperties listenToServiceAndCluster(String serviceName)
     }
     catch (TimeoutException e)
     {
+      ServiceProperties serviceProperties = getServicePropertiesFromCache(serviceName, servicePropertiesFutureCallback);
+      if (serviceProperties != null)
+      {
+        return serviceProperties;
+      }
```

Comment:
If the property (even an old version) is already in the cache, the timeout wouldn't be hit? This shouldn't help.

Reply:
the timeout happens when getting data from event bus (thus the backend -- INDIS or ZK). The xDSClientImpl is cutting the timeout when the response data is invalid or removed. 

The cache (simple load balancer) is just a bottom line to serve the request, not related to killing timeout.